File: crates/app/frontend-routes/api/monitor.ts
import request from '@/utils/http'

// ─── 服务监控 ────────────────────────────────────────────────────────────────

/** 获取服务器信息 */
export function fetchGetServerInfo() {
  return request.get<Api.Monitor.ServerInfoVo>({
    url: '/api/monitor/server'
  })
}

// ─── 缓存监控 ────────────────────────────────────────────────────────────────

/** 获取缓存信息 */
export function fetchGetCacheInfo() {
  return request.get<Api.Monitor.CacheInfoVo>({
    url: '/api/monitor/cache/info'
  })
}

/** 获取缓存键列表 */
export function fetchGetCacheKeys(params?: Api.Monitor.CacheKeysQuery) {
  return request.get<Api.Monitor.CacheKeysVo>({
    url: '/api/monitor/cache/keys',
    params
  })
}

/** 获取缓存键详情 */
export function fetchGetCacheKeyDetail(key: string) {
  return request.get<Api.Monitor.CacheKeyDetailVo>({
    url: `/api/monitor/cache/keys/${encodeURIComponent(key)}/value`
  })
}

/** 删除缓存键 */
export function fetchDeleteCacheKey(key: string) {
  return request.del<null>({
    url: `/api/monitor/cache/keys/${encodeURIComponent(key)}`
  })
}

/** 批量删除缓存键（按模式匹配） */
export function fetchDeleteCacheKeysByPattern(params: Api.Monitor.CacheDeleteQuery) {
  return request.del<null>({
    url: '/api/monitor/cache/keys',
    params
  })
}

File: crates/summer-system/src/router/monitor.rs
use summer_admin_macros::log;
use summer_common::error::ApiResult;
use summer_common::extractor::{Path, Query};
use summer_common::response::Json;
use summer_system_model::dto::monitor::{CacheDeleteQuery, CacheKeysQuery};
use summer_system_model::vo::monitor::{CacheInfoVo, CacheKeyDetailVo, CacheKeysVo, ServerInfoVo};
use summer_web::extractor::Component;
use summer_web::{delete_api, get_api};

use crate::service::monitor_service::{CacheMonitorService, ServerMonitorService};

// ─── 服务监控 ────────────────────────────────────────────────────────────────

#[log(module = "服务监控", action = "查询服务器信息", biz_type = Query)]
#[get_api("/monitor/server")]
pub async fn server_info(
    Component(svc): Component<ServerMonitorService>,
) -> ApiResult<Json<ServerInfoVo>> {
    let vo = svc.get_server_info().await?;
    Ok(Json(vo))
}

// ─── 缓存监控 ────────────────────────────────────────────────────────────────

#[log(module = "缓存监控", action = "查询缓存信息", biz_type = Query)]
#[get_api("/monitor/cache/info")]
pub async fn cache_info(
    Component(svc): Component<CacheMonitorService>,
) -> ApiResult<Json<CacheInfoVo>> {
    let vo = svc.get_cache_info().await?;
    Ok(Json(vo))
}

#[log(module = "缓存监控", action = "查询缓存键列表", biz_type = Query)]
#[get_api("/monitor/cache/keys")]
pub async fn cache_keys(
    Component(svc): Component<CacheMonitorService>,
    Query(query): Query<CacheKeysQuery>,
) -> ApiResult<Json<CacheKeysVo>> {
    let vo = svc.get_cache_keys(query).await?;
    Ok(Json(vo))
}

#[log(module = "缓存监控", action = "查询缓存键详情", biz_type = Query)]
#[get_api("/monitor/cache/keys/{key}/value")]
pub async fn cache_key_detail(
    Component(svc): Component<CacheMonitorService>,
    Path(key): Path<String>,
) -> ApiResult<Json<CacheKeyDetailVo>> {
    let vo = svc.get_cache_key_detail(&key).await?;
    Ok(Json(vo))
}

#[log(module = "缓存监控", action = "删除缓存键", biz_type = Delete)]
#[delete_api("/monitor/cache/keys/{key}")]
pub async fn delete_cache_key(
    Component(svc): Component<CacheMonitorService>,
    Path(key): Path<String>,
) -> ApiResult<()> {
    svc.delete_cache_key(&key).await?;
    Ok(())
}

#[log(module = "缓存监控", action = "批量删除缓存键", biz_type = Delete)]
#[delete_api("/monitor/cache/keys")]
pub async fn delete_cache_keys_by_pattern(
    Component(svc): Component<CacheMonitorService>,
    Query(query): Query<CacheDeleteQuery>,
) -> ApiResult<()> {
    svc.delete_cache_keys_by_pattern(query).await?;
    Ok(())
}
